packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergBridgeJS2Parent.java

```diff
@@ -28,17 +28,19 @@ interface MediaSelectedCallback {
     }
 
     interface MediaUploadEventEmitter {
-        void onUploadMediaFileClear(int mediaId);
+        void onUploadMediaFileClear(int mediaId, float progress);
         void onMediaFileUploadProgress(int mediaId, float progress);
         void onMediaFileUploadSucceeded(int mediaId, String mediaUrl, int serverId, WritableNativeMap metadata);
-        void onMediaFileUploadFailed(int mediaId);
+        void onMediaFileUploadFailed(int mediaId, float progress);
+        void onMediaFileUploadPaused(int mediaId, float progress);
     }
 
     interface MediaSaveEventEmitter {
-        void onSaveMediaFileClear(String mediaId);
+        void onSaveMediaFileClear(String mediaId, float progress);
         void onMediaFileSaveProgress(String mediaId, float progress);
         void onMediaFileSaveSucceeded(String mediaId, String mediaUrl);
-        void onMediaFileSaveFailed(String mediaId);
+        void onMediaFileSaveFailed(String mediaId, float progress);
```

Comment:
Were the `float progress` additions in the name of consistency? I noted the Android implementation has a dedicated progress event/method. 

https://github.com/WordPress/gutenberg/blob/0aabb54e931fb4652bbe22d1fc03764dc69b55c2/packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/WPAndroidGlue/WPAndroidGlueCode.java#L1145-L1147

I.e., I wondered if Android is currently implemented in a way that negates the need to pass progress to the other methods. Should we avoid adding these `float progress`?

iOS is implemented very differently and does not have a dedicated progress event/method. iOS relies upon a enum state that is passed to a single [media observer method](https://github.com/wordpress-mobile/WordPress-iOS/blob/517584306972888d9ddc912f274d0d09046cde46/WordPress/Classes/ViewRelated/Gutenberg/GutenbergMediaInserterHelper.swift#L198-L266) with a switch statement that communicates different states/progresses to the JavaScript app. The single method can receive a [progress event](https://github.com/wordpress-mobile/WordPress-iOS/blob/517584306972888d9ddc912f274d0d09046cde46/WordPress/Classes/ViewRelated/Gutenberg/GutenbergMediaInserterHelper.swift#L262-L264) with a value.